controllers/v1/programs.js

```diff
@@ -3,13 +3,385 @@
  * author : vishnu
  * created-date : 09-Mar-2022
  * Description : programs related information.
- */
+*/
+
+
+// Dependencies
+const programsHelper = require(MODULES_BASE_PATH + "/programs/helper")
+
+
 module.exports = class  Programs extends Abstract{
-    constructor() {
-      super("programs");
+  constructor() {
+    super("programs");
+  }
+  static get name() {
+    return "programs";
+  }
+    
+
+   /**
+   * Create program.
+   * @method
+   * @name create
+   * @param {Object} req - requested data.
+   * @returns {JSON} - created program document.
+   */
+
+    async create(req) {
+      return new Promise(async (resolve, reject) => {
+        try {
+          let programCreationData = await programsHelper.create(
+            req.body,
+            req.userDetails.userId,
+            true
+          );
+          
+          return resolve(programCreationData)
+  
+        }
+        catch (error) {
+          reject({
+            status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+            message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+            errorObject: error
+          })
+        }
+      })
+    }
+
+
+  /**
+   * Update program.
+   * @method
+   * @name update
+   * @param {Object} req - requested data.
+   * @apiParamExample {json} Request-Body:
+   * {
+      "externalId" : "PROGID01",
+      "name" : "DCPCR School Development Index 2018-19",
+      "description" : "DCPCR School Development Index 2018-19",
+      "isDeleted" : false,
+      "resourceType" : [ 
+          "program"
+      ],
+      "language" : [ 
+          "English"
+      ],
+      "keywords" : [],
+      "concepts" : [],
+      "userId":"a082787f-8f8f-42f2-a706-35457ca6f1fd",
+      "imageCompression" : {
+          "quality" : 10
+      },
+      "components" : [ 
+          "5b98fa069f664f7e1ae7498c"
+      ],
+      "scope" : {
+          "entityType" : "state",
+          "entities" : ["bc75cc99-9205-463e-a722-5326857838f8","8ac1efe9-0415-4313-89ef-884e1c8eee34","5f33c3d85f637784791cd830"],
+          "roles" : ["HM","BEO"]
+      },
+      "requestForPIIConsent" : true
+  }
+
+   * @param {Object} 
+   * @returns {JSON} - 
+   */
+
+  async update(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let programUpdationData = await programsHelper.update(
+          req.params._id,
+          req.body,
+          req.userDetails.userId,
```

Comment:
@Prajwal17Tunerlabs do you get userId from req.userDetails ? In authenticator you have added userId in userInformation